clippy_lints/src/methods/mod.rs

```diff
@@ -1773,7 +1773,7 @@ fn derefs_to_slice<'a, 'tcx>(
     expr: &'tcx hir::Expr,
     ty: Ty<'tcx>,
 ) -> Option<&'tcx hir::Expr> {
-    fn may_slice(cx: &LateContext<'_, '_>, ty: Ty<'_>) -> bool {
+    fn may_slice<'a>(cx: &LateContext<'_, 'a>, ty: Ty<'a>) -> bool {
```

Comment:
I'm not sure, but shouldn't the `Ty` lifetime be called `'tcx`?

Reply:
Yes, but does it conflict with outer function?

Reply:
Oh, I didn't see that it was an inner function. I think then it's fine =)